### PR TITLE
Added the ability to skip redownload when marking an item as failed in the activity queue

### PR DESCRIPTION
**Pull request by PearsonFlyer**
This code is lifted from Lidarr. When marking an item as failed in the activity queue, a new check-box appears allowing you to skip a re-download search, and just mark it as failed and remove it from the queue instead.

Here is what it looks like before you check the blocklist box:

image

And here is what it looks like when you check the blocklist box:

image

Also added the use of translations for these functions.

#### Database Migration
NO

#### Description
Sometimes when you're blocklisting an item, you don't want it to re-search and download the next thing.

#### Todos
- [ ] Tests
- [ ] Wiki Updates

#### Issues Fixed or Closed by this PR

*

### CODE REVIEW

File: frontend/src/Activity/Queue/QueueRow.js [hunk 1]
```diff
@@ -44,14 +44,15 @@ class QueueRow extends Component {
     this.setState({ isRemoveQueueItemModalOpen: true });
   };
 
-  onRemoveQueueItemModalConfirmed = (blocklist) => {
+  onRemoveQueueItemModalConfirmed = (blocklist, skipredownload) => {
```
Comment (markus101) [inline frontend/src/Activity/Queue/QueueRow.js:47]: This differs from Lidarr, but Lidarr's variable should be renamed. `skipRedownload` (several places this should be changed)

File: src/NzbDrone.Core/Download/DownloadFailedEvent.cs [hunk 2]
```diff
@@ -23,5 +23,6 @@ public DownloadFailedEvent()
         public Dictionary<string, string> Data { get; set; }
         public TrackedDownload TrackedDownload { get; set; }
         public List<Language> Languages { get; set; }
+        public bool SkipReDownload { get; set; }
```
Comment (markus101) [inline src/NzbDrone.Core/Download/DownloadFailedEvent.cs:26]: `SkipRedownload` is more correct.

**Review by markus101 — CHANGES_REQUESTED** [review]
The original Lidarr work has some incorrect variable naming that should probably be resolved, but should be fixed in this PR before landing.

**Comment by jack-mil** [comment]
I like this feature! 
I can also imagine a use case where a user wants to skip a redownload search, but not necessarily add the release to a blocklist. This could be to perform a manual interactive search and select a specific release.

Could the option to skip redownload search be decoupled from adding the release to the blocklist ?

**Comment by PearsonFlyer** [comment]
All case changes made as requested.

File: frontend/src/Activity/Queue/RemoveQueueItemsModal.js [hunk 3]
```diff
@@ -122,6 +128,19 @@ class RemoveQueueItemsModal extends Component {
               />
             </FormGroup>
 
+            {
+              blocklist &&
```
Comment (markus101) [inline frontend/src/Activity/Queue/RemoveQueueItemsModal.js:132]: Please convert to a ternary: `blocklist ? ... : null`, similar to `allPending` starting at line 100 above.

**Review by markus101 — COMMENTED** [review]
Just one more small thing to change

File: package.json [hunk 4]
```diff
@@ -101,7 +101,7 @@
     "@types/react-router-dom": "5.3.3",
     "@types/react-text-truncate": "0.14.1",
     "@types/react-window": "1.8.5",
-    "@types/redux-actions": "2.6.2",
```
Comment (markus101) [inline package.json:104]: This change needs to be reverted

Comment (PearsonFlyer) [reply package.json]: Done.

**Comment by PearsonFlyer** [comment]
I merged the develop branch en.json file with my 2 new additions, which should resolve conflicts there.

**Comment by markus101** [comment]
The build is failing with some linting issues: https://build.sonarr.tv/buildConfiguration/Sonarr_V4PullRequests_1BuildFrontend/200669?buildTab=log&logView=flowAware&linesState=86.113&focusLine=113

`yarn lint` locally will show the errors as well.

**Comment by PearsonFlyer** [comment]
> The build is failing with some linting issues: https://build.sonarr.tv/buildConfiguration/Sonarr_V4PullRequests_1BuildFrontend/200669?buildTab=log&logView=flowAware&linesState=86.113&focusLine=113
> 
> `yarn lint` locally will show the errors as well.

Weird, I do lint before building. I am not getting any lint errors at all.

`E:\repos\sonarr\src>yarn lint
yarn run v1.22.19
$ eslint --config frontend/.eslintrc.js --ignore-path frontend/.eslintignore frontend/
Done in 19.07s.`

**Comment by markus101** [comment]
Odd, looks like `en.json` has a conflict, so maybe after rebasing it'll be good.

**Comment by markus101** [comment]
Turns out trying to fix merge conflicts from GitHub is a disaster, fixed everything up locally and landed in d7025a98deb321397db9a4933cf688b68eafeba9

**Pull request closed without merging**